src/tvgFlutterLottieAnimation.cpp

```diff
@@ -0,0 +1,311 @@
+#include <thorvg.h>
+#include "tvgFlutterLottieAnimation.h"
+
+using namespace std;
+using namespace tvg;
+
+// NOTE: build error with `const char*`
+static char *NoError = "None";
+
+class __attribute__((visibility("default"))) TvgLottieAnimation
+{
+public:
+    ~TvgLottieAnimation()
+    {
+        free(buffer);
+        Initializer::term(CanvasEngine::Sw);
+    }
+
+    static unique_ptr<TvgLottieAnimation> create()
+    {
+        return unique_ptr<TvgLottieAnimation>(new TvgLottieAnimation());
+    }
+
+    char *error()
+    {
+        return errorMsg;
+    }
+
+    float *size()
+    {
+        return psize;
+    }
+
+    float duration()
+    {
+        if (!canvas || !animation)
+            return 0;
+        return animation->duration();
+    }
+
+    float totalFrame()
+    {
+        if (!canvas || !animation)
+            return 0;
+        return animation->totalFrame();
+    }
+
+    float curFrame()
+    {
+        if (!canvas || !animation)
+            return 0;
+        return animation->curFrame();
+    }
+
+    void resize(int w, int h)
+    {
+        if (!canvas || !animation)
+            return;
+        if (width == w && height == h)
+            return;
+
+        width = w;
+        height = h;
+
+        free(buffer);
+        buffer = (uint8_t *)malloc(width * height * sizeof(uint32_t));
+        canvas->target((uint32_t *)buffer, width, width, height, SwCanvas::ABGR8888S);
```

Comment:
I'm a bit surprised that this happens within this function... it's unintuitive that a resize would set the target canvas